### PR TITLE
Nessie GC: Command line tool

### PR DESCRIPTION
**Pull request by snazy**
Adds a command-line tool for Nessie GC.

Depeends on #5207

### CODE REVIEW

File: gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/commands/JdbcCreateSchema.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.projectnessie.gc.contents.jdbc.JdbcHelper;
+import org.projectnessie.gc.tool.cli.Closeables;
+import org.projectnessie.gc.tool.cli.options.EnvironmentDefaultProvider;
+import org.projectnessie.gc.tool.cli.options.JdbcOptions;
+import picocli.CommandLine;
+import picocli.CommandLine.Help.Ansi;
+import picocli.CommandLine.Model.CommandSpec;
+
+@CommandLine.Command(
+    name = "show-sql-create-schema-script",
```
Comment (ajantha-bhat) [inline gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/commands/JdbcCreateSchema.java:29]: "show" is confusing for me as it does "create" function also.

File: gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/commands/BaseRepositoryCommand.java [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+  protected int expire(
+      LiveContentSetsRepository liveContentSetsRepository,
+      LiveContentSet liveContentSet,
+      SweepOptions sweepOptions,
+      IcebergOptions icebergOptions,
+      CommandSpec commandSpec) {
+    if (liveContentSet.status() != LiveContentSet.Status.IDENTIFY_SUCCESS) {
+      throw new ExecutionException(
+          commandSpec.commandLine(),
+          "Expected live-set to have status IDENTIFY_SUCCESS, but status is "
```
[the remark this answers is not in the document]
Comment (snazy) [reply gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/commands/BaseRepositoryCommand.java:216]: user readable error

File: gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/CLI.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import org.projectnessie.gc.tool.cli.commands.MarkAndSweep;
+import org.projectnessie.gc.tool.cli.commands.MarkLive;
+import org.projectnessie.gc.tool.cli.commands.ShowLiveSet;
+import org.projectnessie.gc.tool.cli.commands.Sweep;
+import picocli.CommandLine;
+import picocli.CommandLine.HelpCommand;
+
+/**
+ * Standalone tool to perform a full {@link IdentifyLiveContents mark}-and-{@link
+ * PerContentDeleteExpired sweep} Nessie GC cycle.
```
[the remark this answers is not in the document]
Comment (snazy) [reply gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/CLI.java:40]: Let's keep it as is for now (that's part of gc-base)

**Comment by snazy** [comment]
Remember: this is a _draft_ PR that depends on #5207

File: gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/options/MarkOptions.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                      ex);
+                })
+            .collect(Collectors.toList());
+
+    return namedReference ->
+        policies.stream()
+            .filter(e -> e.getKey().test(namedReference.getName()))
+            .map(Entry::getValue)
+            .findFirst()
+            .orElse(CutoffPolicy.NONE);
```
Comment (ajantha-bhat) [inline gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/options/MarkOptions.java:136]: While playing with the tool. I felt we need to have default cutoff time configurations.

For example, If I have 100 branch that needs gc, I can't pass each branch name with cutoff time to the cutoff argument.

File: gc/gc-tool-inttest/src/test/java/org/projectnessie/gc/tool/inttest/ITSparkIcebergNessieCLI.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    System.setProperty("aws.region", "us-east-1");
+    System.setProperty("aws.s3.endpoint", minio.s3endpoint());
+    System.setProperty("aws.s3.accessKey", minio.accessKey());
+    System.setProperty("aws.s3.secretAccessKey", minio.secretKey());
+
+    return r;
+  }
+
+  @Test
+  public void fullRoundTrip(@TempDir Path workDir) throws Exception {
```
Comment (ajantha-bhat) [inline gc/gc-tool-inttest/src/test/java/org/projectnessie/gc/tool/inttest/ITSparkIcebergNessieCLI.java:81]: I found an issue in this scenario

Have three commits, execute gc with defer-deletes and cutoff policy with retain 1 commit. Then delete the deferred deletes. Then execute gc again without defer option and it throws `FileNotFoundException`. It is probably Identifying the expired contents again from Nessie (this problem not happening for in-memory!)

Detailed steps:

create table nessie.db1.t1(id int) using iceberg;
insert into nessie.db1.t1 select 42;
insert into nessie.db1.t1 select 42;

./nessie-gc gc --defer-deletes --cutoff main=1 --jdbc-url jdbc:postgresql://127.0.0.1:5432/nessie_gc --jdbc-user pguser --jdbc-password mysecretpassword

./nessie-gc deferred-deletes -l 425c66b2-7e72-496d-b93d-07e96bd5dfc5  --jdbc-url jdbc:postgresql://127.0.0.1:5432/nessie_gc --jdbc-user pguser --jdbc-password mysecretpassword

./nessie-gc gc  --jdbc-url jdbc:postgresql://127.0.0.1:5432/nessie_gc --jdbc-user pguser --jdbc-password mysecretpassword


org.apache.iceberg.exceptions.NotFoundException: Failed to open input stream for file: /tmp/nessie2/db1/t1/metadata/00000-397fa47a-eb9a-477f-bcf3-4453dfde84fb.metadata.json
	at org.apache.iceberg.hadoop.HadoopInputFile.newStream(HadoopInputFile.java:184)
	at org.apache.iceberg.TableMetadataParser.read(TableMetadataParser.java:263)
	at org.apache.iceberg.TableMetadataParser.read(TableMetadataParser.java:258)
	at org.projectnessie.gc.iceberg.IcebergContentToFiles.extractFiles(IcebergContentToFiles.java:70)
	at org.projectnessie.gc.expire.PerContentDeleteExpired.lambda$identifyLiveFiles$2(PerContentDeleteExpired.java:125)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
	at org.projectnessie.gc.contents.jdbc.JdbcHelper$ResultSetSplit.tryAdvance(JdbcHelper.java:115)
	at java.base/java.util.Spliterator.forEachRemaining(Spliterator.java:326)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.ReduceOps$5.evaluateSequential(ReduceOps.java:257)
	at java.base/java.util.stream.ReduceOps$5.evaluateSequential(ReduceOps.java:248)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.count(ReferencePipeline.java:605)
	at org.projectnessie.gc.expire.PerContentDeleteExpired.identifyLiveFiles(PerContentDeleteExpired.java:134)
	at org.projectnessie.gc.expire.PerContentDeleteExpired.expire(PerContentDeleteExpired.java:73)
	at org.projectnessie.gc.expire.local.DefaultLocalExpire.expireSingleContent(DefaultLocalExpire.java:104)

Comment (ajantha-bhat) [reply gc/gc-tool-inttest/src/test/java/org/projectnessie/gc/tool/inttest/ITSparkIcebergNessieCLI.java:81]: I just narrowed it down into simple steps now and reproduced it with the in-memory itself.

```
create table nessie.db1.t1(id int) using iceberg;
insert into nessie.db1.t1 select 42;
insert into nessie.db1.t1 select 42;

./nessie-gc gc --cutoff main=1 --inmemory

./nessie-gc gc --cutoff --inmemory
```

**Review by ajantha-bhat — APPROVED** [review]

**Review by ajantha-bhat — APPROVED** [review]
Please update the PR description.

**Pull request merged**